test/unit_tests/braket/circuits/noise/test_unitary_gate_criteria.py

```diff
@@ -0,0 +1,215 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import numpy as np
+import pytest
+
+from braket.circuits import Gate, Instruction, Observable
+from braket.circuits.gates import Unitary
+from braket.circuits.noise_model import (
+    CriteriaKey,
+    CriteriaKeyResult,
+    ObservableCriteria,
+    UnitaryGateCriteria,
+)
+
+
+def h_unitary():
+    return Unitary(Gate.H().to_matrix())
+
+
+def i_unitary():
+    return Unitary(Gate.I().to_matrix())
+
+
+def x_unitary():
+    return Unitary(Gate.X().to_matrix())
+
+
+def cnot_unitary():
+    return Unitary(Gate.CNot().to_matrix())
+
+
+@pytest.mark.parametrize(
+    "unitary, qubits",
+    [
+        (h_unitary(), range(3)),
+        (h_unitary(), 1),
+        (h_unitary(), None),
+    ],
+)
+def test_happy_case(unitary, qubits):
+    criteria = UnitaryGateCriteria(unitary=unitary, qubits=qubits)
+    assert criteria.applicable_key_types() == [
+        CriteriaKey.QUBIT,
+        CriteriaKey.UNITARY_GATE,
+    ]
+    assert h_unitary().to_matrix().tobytes() in criteria.get_keys(CriteriaKey.UNITARY_GATE)
+    if qubits is None:
+        assert CriteriaKeyResult.ALL == criteria.get_keys(CriteriaKey.QUBIT)
+    else:
+        assert 1 in criteria.get_keys(CriteriaKey.QUBIT)
+
+
+def test_serialization():
+    test_criteria = UnitaryGateCriteria(unitary=h_unitary(), qubits=range(3))
+    serialized_criteria = test_criteria.to_dict()
+    assert serialized_criteria["__class__"] == "UnitaryGateCriteria"
+    assert "unitary" in serialized_criteria
+    assert "qubits" in serialized_criteria
+    deserialized_criteria = UnitaryGateCriteria.from_dict(serialized_criteria)
+    assert test_criteria == deserialized_criteria
+
+
+@pytest.mark.parametrize(
+    "unitary, qubits, defaults, matching_instructions, non_matching_instructions",
+    [
+        (
+            h_unitary(),
+            range(3),
+            None,
+            [
+                Instruction(h_unitary(), 0),
+                Instruction(h_unitary(), 1),
```

Comment:
0 covers the edge case (first number we support), 1 covers the regular cases (a non-edge case number we support). We can add 2, and other numbers, but they're just more regular cases, so they shouldn't be needed - unless I'm missing something?

Reply:
right, I just figured since we're doing `range(3)`, we should just test it matches all three instructions: ` Instruction(h_unitary(), i)` for `i = 0,1,2` to be consistent. Otherwise why not just use `range(2)`?